src/makeAnimatedComponent.lua

```diff
@@ -85,8 +85,10 @@ local function makeAnimatedComponent(toWrap)
 			self._rbx = rbx
 
 			-- forward the ref to the original component
-			if ref then
+			if ref and typeof(ref) == "function" then
```

Comment:
Nice! One thought on this...could it be moved into a pair of if statements, something like:

```lua
if ref then
    if typeof(ref) == "function" then
       -- ...
```

That will cut down on code duplication!

Reply:
oh, smart idea! i've always tried to use as little nesting as possible, though i realize how not repeating checks could make the code more readable (: